quipucords/cli/network/add.py

```diff
@@ -54,6 +56,13 @@ def __init__(self, subparsers):
     def _validate_args(self):
         CliCommand._validate_args(self)
 
+        if self.args.hosts and len(self.args.hosts) == 1:
+            # check if a file and read in values
+            filename = self.args.hosts[0]
+            input_path = os.path.expanduser(os.path.expandvars(filename))
```

Comment:
It looks like you put the `os.path.expanduser(os.path.expandvars(filename))` inside `read_in_file`, but I see that you still need this because of the `os.path.isfile()`. Can we change the interface so we don't have to duplicate this code?

Maybe `read_in_file` could raise a special exception if the path is not a file, and this code could catch that exception and keep going. Or maybe return `None` if the input's not a file, if that's easier.